src/libsystemd-network/sd-dhcp-client.c

```diff
@@ -239,6 +241,25 @@ int sd_dhcp_client_set_callback(
         return 0;
 }
 
+int sd_dhcp_client_set_state_callback(
+                sd_dhcp_client *client,
+                sd_dhcp_client_callback_t cb,
+                void *userdata) {
+
+        assert_return(client, -EINVAL);
+
+        client->state_callback = cb;
+        client->state_userdata = userdata;
+
+        return 0;
+}
+
+int sd_dhcp_client_get_state( sd_dhcp_client *client) {
+        assert(client);
+
+        return client->state;
```

Comment:
Hmm. The state enum is not exposed in `sd-dhcp-client.h`.
Please move the declaration of the state enum to sd-dhcp-client.h.

Reply:
Done.